Dockerfile

```
@@ -0,0 +1,17 @@
+FROM python:3.11-slim-buster as base
+
+RUN apt-get update && apt-get install -y curl
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="${PATH}:/root/.local/share/pypoetry/venv/bin/"
+
+COPY pyproject.toml /pyproject.toml
```

In addition to the `pyproject.toml` which specifies our top level dependencies, it's a good habit to also copy in the `poetry.lock` so that we're more confident that we're consistently using the same libraries throughout the application
```suggestion
COPY pyproject.toml /pyproject.toml
COPY poetry.lock /poetry.lock
```

---

Separately, this works but I find having things a bit more compartmentalised helpful, so e.g. rather than just putting this file in the root folder we might want to:
* set a working directory to hold our application code
* copy into that, potentially using relative paths so we don't need to update it if we change that later
```docker
WORKDIR /app
COPY pyproject.toml .
...
```